src/modules/users/components/TokenActivation/TokenActivationContent/ChangeTokenStateForm.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useCallback, useState } from 'react';
+import { FormattedMessage, defineMessages } from 'react-intl';
+import { BigNumber, bigNumberify } from 'ethers/utils';
+import { FormikProps } from 'formik';
+import moveDecimal from 'move-decimal-point';
+import * as yup from 'yup';
+
+import Button from '~core/Button';
+import { ActionForm, Input } from '~core/Fields';
+import Numeral from '~core/Numeral';
+
+import { UserToken } from '~data/generated';
+import { ActionTypes } from '~redux/index';
+import { pipe, mapPayload } from '~utils/actions';
+
+import styles from './TokenActivationContent.css';
+
+const MSG = defineMessages({
+  changeState: {
+    id: 'users.TokenActivation.TokenActivationContent.TokensTab.changeState',
+    defaultMessage: 'Change token state',
+  },
+  activate: {
+    id: 'users.TokenActivation.TokenActivationContent.TokensTab.activate',
+    defaultMessage: 'Activate',
+  },
+  withdraw: {
+    id: 'users.TokenActivation.TokenActivationContent.TokensTab.withdraw',
+    defaultMessage: 'Withdraw',
+  },
+  balance: {
+    id: 'users.TokenActivation.TokenActivationContent.TokensTab.balance',
+    defaultMessage: 'balance: {tokenBalance}',
+  },
+});
+
+const validationSchema = yup.object({
+  amount: yup.number().required().moreThan(0),
+});
+
+type FormValues = {
+  amount: number;
+};
+
+export interface ChangeTokenStateFormProps {
+  token: UserToken;
+  tokenDecimals: number;
+  activeTokens: BigNumber;
+  inactiveTokens: BigNumber;
+}
+
+const ChangeTokenStateForm = ({
+  token,
+  tokenDecimals,
+  activeTokens,
+  inactiveTokens,
+}: ChangeTokenStateFormProps) => {
+  const [isActivate, setIsActivate] = useState(true);
+
+  const formAction = useCallback(
+    (actionType: '' | '_ERROR' | '_SUCCESS') =>
+      isActivate
+        ? ActionTypes[`USER_DEPOSIT_TOKEN${actionType}`]
+        : ActionTypes[`USER_WITHDRAW_TOKEN${actionType}`],
+    [isActivate],
+  );
+
+  const transform = useCallback(
+    pipe(
+      mapPayload(({ amount }) => {
+        // Convert amount string with decimals to BigInt (eth to wei)
+        const formtattedAmount = bigNumberify(
+          moveDecimal(amount, tokenDecimals),
+        );
+
+        return { amount: formtattedAmount };
```

Comment:
don't you need to send (at least) the colony address as well ?

Without it, the redux transaction channel doesn't know for what colony to instantiate the client for.

Now that I think about it more, should this also have the user address? Since the transaction will be made for that user.